src/usr/local/www/firewall_rules_edit.php

```diff
@@ -42,8 +42,8 @@
 }
 
 function is_posnumericint($arg) {
-	// Note that to be safe we do not allow any leading zero - "01", "007"
-	return (is_numericint($arg) && $arg[0] != '0' && $arg > 0);
+	// Integer > 0? (Note that to be safe we do not allow any leading zero - "01", "007")
+	return (is_numericint($arg) && $arg[0] != '0');
```

Comment:
is_numericint() only passes integers >= 0 or non-empty strings containing 0-9 only. So there's no need to test again for negatives, and since empty strings are excluded, if the first character isn't zero, the value can't be zero either.